topics/data-science/tutorials/python-typing/tutorial.md

```diff
@@ -0,0 +1,263 @@
+---
+layout: tutorial_hands_on
+
+title: Python - Typing
+level: Intermediate
+requirements: []
+follow_up_training: []
+questions:
+- What is typing?
+- How does it improve code?
+- Can it help me?
+
+objectives:
+- Understand the utility of annotating types on one's code
+- Understand the limits of typing in python
+
+time_estimation:  30M
+key_points:
+- Typing improves the correctness and quality of your code
+- It can ensure that editor provided hints are better and more accurate.
+
+subtopic: python-modular
+contributions:
+  authorship:
+  - hexylena
+  editing:
+  - dirowa
+  - bazante1
+
+priority: 3
+notebook:
+  language: python
+---
+
+In some, better, languages typing is a core part of the language and types are checked at compile time, to ensure your code can never use the incorrect type of object. Python, and a few other dynamic languages, instead use "Duck Typing" wherein the type of the object is less important than whether or not the correct methods or attributes are available.
+
+However, we can provide type hints as we write python which will allow our editor to type check code as we go, even if it is not enforced at any point.
```

Comment:
```suggestion
However, we can provide type hints as we write python which will allow our editor to type check code as we go, even if it is not typically enforced at any point.
```

there are tools in the ecosystem that can enforce type annotations at runtime

Reply:
really? Huh, til.